fido/src/main/java/com/yubico/yubikit/fido/ctap/FingerprintBioEnrollment.java

```diff
@@ -0,0 +1,496 @@
+    private static final int PARAM_TIMEOUT_MS = 0x03;
+
+    public static final int FEEDBACK_FP_GOOD = 0x00;
+    public static final int FEEDBACK_FP_TOO_HIGH = 0x01;
+    public static final int FEEDBACK_FP_TOO_LOW = 0x02;
+    public static final int FEEDBACK_FP_TOO_LEFT = 0x03;
+    public static final int FEEDBACK_FP_TOO_RIGHT = 0x04;
+    public static final int FEEDBACK_FP_TOO_FAST = 0x05;
+    public static final int FEEDBACK_FP_TOO_SLOW = 0x06;
+    public static final int FEEDBACK_FP_POOR_QUALITY = 0x07;
+    public static final int FEEDBACK_FP_TOO_SKEWED = 0x08;
+    public static final int FEEDBACK_FP_TOO_SHORT = 0x09;
+    public static final int FEEDBACK_FP_MERGE_FAILURE = 0x0A;
+    public static final int FEEDBACK_FP_EXISTS = 0x0B;
+    // 0x0C not used
+    public static final int FEEDBACK_NO_USER_ACTIVITY = 0x0D;
+    public static final int FEEDBACK_NO_UP_TRANSITION = 0x0E;
+
+    private final PinUvAuthProtocol pinUvAuth;
+    private final byte[] pinUvToken;
+    private final SensorInfo cachedSensorInfo;
+
+    private final org.slf4j.Logger logger = LoggerFactory.getLogger(FingerprintBioEnrollment.class);
+
+    public static class SensorInfo {
+        public final int fingerprintKind;
+        public final int maxCaptureSamplesRequiredForEnroll;
+        public final int maxTemplateFriendlyName;
+
+        public SensorInfo(int fingerprintKind, int maxCaptureSamplesRequiredForEnroll, int maxTemplateFriendlyName) {
+            this.fingerprintKind = fingerprintKind;
+            this.maxCaptureSamplesRequiredForEnroll = maxCaptureSamplesRequiredForEnroll;
+            this.maxTemplateFriendlyName = maxTemplateFriendlyName;
+        }
+
+        /**
+         * Indicates type of fingerprint sensor.
+         *
+         * @return For touch type fingerprints returns 1, for swipe type fingerprints returns 2.
+         */
+        public int getFingerprintKind() {
+            return fingerprintKind;
+        }
+
+        /**
+         * Indicates the maximum good samples required for enrollment.
+         *
+         * @return Maximum good samples required for enrollment.
+         */
+        public int getMaxCaptureSamplesRequiredForEnroll() {
+            return maxCaptureSamplesRequiredForEnroll;
+        }
+
+        /**
+         * Indicates the maximum number of bytes the authenticator will accept as a templateFriendlyName.
+         *
+         * @return Maximum number of bytes the authenticator will accept as a templateFriendlyName.
+         */
+        public int getMaxTemplateFriendlyName() {
+            return maxTemplateFriendlyName;
+        }
+    }
+
+    public static class CaptureError extends Exception {
+        private final int code;
+
+        public CaptureError(int code) {
+            super("Fingerprint capture error: " + code);
+            this.code = code;
+        }
+
+        public int getCode() {
+            return code;
+        }
+    }
+
+    static class CaptureStatus {
```

Comment:
Is this (and EnrollBeginStatus below) OK to be package private? Isn't it returned from public methods?

Reply:
fixed this 👍🏻